src/Bicep.Core/TypeSystem/ParamsTypeAssignmentVisitor.cs

```diff
@@ -0,0 +1,294 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Semantics;
+using Bicep.Core.Syntax;
+using System.Linq;
+using Bicep.Core.Extensions;
+using System.Collections.Concurrent;
+
+namespace Bicep.Core.TypeSystem
+{
+    public sealed class ParamsTypeAssignmentVisitor : SyntaxVisitor
+    {
+        private class ParamsTypeAssignment
```

Comment:
This is the same as the `TypeAssignment` class in `TypeAssignmentVisitor`, so it's probably worth making a `TypeAssignment` public and reuse it in `ParamsTypeAssignmentVisitor`.